src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline

```diff
@@ -7586,9 +7587,9 @@ public class LibraryClient implements BackgroundResource {
             .setRequiredSingularString(requiredSingularString)
             .setRequiredSingularBytes(requiredSingularBytes)
             .setRequiredSingularMessage(requiredSingularMessage)
-            .setRequiredSingularResourceName(requiredSingularResourceName)
-            .setRequiredSingularResourceNameOneof(requiredSingularResourceNameOneof)
-            .setRequiredSingularResourceNameCommon(requiredSingularResourceNameCommon)
+            .setRequiredSingularResourceName(requiredSingularResourceName == null ? null : requiredSingularResourceName.toString())
```

Comment:
Is it correct that a required resource-name is allowed to be `null`?

Reply:
This has been how java delegates the overload with resource name to the non-flattened method. 

For example: https://github.com/googleapis/java-tasks/blob/master/google-cloud-tasks/src/main/java/com/google/cloud/tasks/v2/CloudTasksClient.java#L1532-L1540

Actually I think protobuf Java does not allow setting null values on any field, so the code will still fail on the client side. I guess we just pass the error handling to protobuf :)